relays/bin-substrate/src/cli/register_parachain.rs

```diff
@@ -0,0 +1,343 @@
+	parachain: Parachain,
+	/// Parachain deposit.
+	#[structopt(long, default_value = "0")]
+	deposit: Balance,
+	/// Lease begin.
+	#[structopt(long, default_value = "0")]
+	lease_begin: u32,
+	/// Lease end.
+	#[structopt(long, default_value = "256")]
+	lease_end: u32,
+	#[structopt(flatten)]
+	relay_connection: RelaychainConnectionParams,
+	#[structopt(flatten)]
+	relay_sign: RelaychainSigningParams,
+	#[structopt(flatten)]
+	para_connection: ParachainConnectionParams,
+}
+
+/// Parachain to register.
+#[derive(Debug, EnumString, EnumVariantNames, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+pub enum Parachain {
+	RialtoParachain,
+}
+
+macro_rules! select_bridge {
+	($bridge: expr, $generic: tt) => {
+		match $bridge {
+			Parachain::RialtoParachain => {
+				type Relaychain = relay_rialto_client::Rialto;
+				type Parachain = relay_rialto_parachain_client::RialtoParachain;
+
+				use bp_rialto::{PARAS_PALLET_NAME, PARAS_REGISTRAR_PALLET_NAME};
+
+				$generic
+			},
+		}
+	};
+}
+
+impl RegisterParachain {
+	/// Run the command.
+	pub async fn run(self) -> anyhow::Result<()> {
+		select_bridge!(self.parachain, {
+			let relay_client = self.relay_connection.to_client::<Relaychain>().await?;
+			let relay_sign = self.relay_sign.to_keypair::<Relaychain>()?;
+			let para_client = self.para_connection.to_client::<Parachain>().await?;
+
+			// hopefully we're the only actor that is registering parachain right now
+			// => read next parachain id
+			let para_id_key = bp_runtime::storage_value_final_key(
+				PARAS_REGISTRAR_PALLET_NAME.as_bytes(),
+				NEXT_FREE_PARA_ID_STORAGE_NAME.as_bytes(),
+			);
+			let para_id: ParaId = relay_client
+				.storage_value(StorageKey(para_id_key.to_vec()), None)
+				.await?
+				.unwrap_or(polkadot_primitives::v1::LOWEST_PUBLIC_ID)
+				.max(polkadot_primitives::v1::LOWEST_PUBLIC_ID);
+			log::info!(target: "bridge", "Going to reserve parachain id: {:?}", para_id);
+
+			// step 1: reserve a parachain id
+			let relay_genesis_hash = *relay_client.genesis_hash();
+			let relay_sudo_account: AccountIdOf<Relaychain> = relay_sign.public().clone().into();
+			let reserve_parachain_id_call: CallOf<Relaychain> = ParaRegistrarCall::reserve().into();
+			let reserve_parachain_signer = relay_sign.clone();
+			wait_until_transaction_is_finalized::<Relaychain>(
+				relay_client
+					.submit_and_watch_signed_extrinsic(
+						relay_sudo_account.clone(),
+						move |_, transaction_nonce| {
+							Bytes(
+								Relaychain::sign_transaction(
+									relay_genesis_hash,
+									&reserve_parachain_signer,
+									relay_substrate_client::TransactionEra::immortal(),
+									UnsignedTransaction::new(
+										reserve_parachain_id_call,
+										transaction_nonce,
+									),
+								)
+								.encode(),
+							)
+						},
+					)
+					.await?,
+			)
+			.await?;
+			log::info!(target: "bridge", "Reserved parachain id: {:?}", para_id);
+
+			// step 2: register parathread
```

Comment:
Couldn't these two steps be batched?

Reply:
We don't have utility pallet in our runtimes